File: leetcode/p00700/p00700.go
package p00700

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

// Time Complexity: O(logn)
// Space Complexity: O(1)
func searchBST(root *TreeNode, val int) *TreeNode {
	current := root
	for current != nil {
		if current.Val == val {
			return current
		} else if current.Val < val {
			current = current.Right
		} else {
			current = current.Left
		}
	}

	return nil
}

File: leetcode/p00700/p00700.py
from typing import Optional

class TreeNode:
    def __ini__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    # Time Complexity: O(logn)
    # Space Complexity: O(1)
    def searchBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        current = root

        while current:
            if current.val == val:
                return current
            elif current.val > val:
                current = current.left
            else:
                current = current.right
        
        return None
